diff --git a/apache/simple/1b48bb3.c b/apache/simple/1b48bb3.c
--- a/apache/simple/1b48bb3.c
+++ b/apache/simple/1b48bb3.c
@@ -8,10 +8,8 @@ void apr_rmm_init(char* rmm_lock)
 #ifdef APU_HAS_LDAP
 void util_ldap_cache_init()
 {
-  char* rmm_lock;
-
 #ifdef APR_HAS_SHARED_MEMORY
-  apr_rmm_init(rmm_lock); // ERROR: rmm_lock uninitialized
+  apr_rmm_init(NULL); // ERROR: rmm_lock uninitialized
 #endif
 }
 #endif
